### PR TITLE
[WIP]windows_dns_client resource

### PR DESCRIPTION
**Pull request by Stromweld**
### Description

This adds a windows resource for configuring a servers interface dns settings. It allows assigning list of static IP's for DNS search servers and/or a list of DNS domain suffixes to append to short names when searching.  

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by Stromweld** [comment]
@tas50 here is that dns_client resource I mentioned at chef conf for possible integration with core chef as well.

File: kitchen.appveyor.yml [hunk 1]
```diff
@@ -16,7 +16,8 @@ provisioner:
   deprecations_as_errors: true
   product_name: chef
   channel: stable
-  version: 14.6
+  version: 15
```
Comment (tas50) [inline kitchen.appveyor.yml]: This should stay at the lowest version supported

File: README.md [hunk 2]
```diff
@@ -9,12 +9,14 @@ This cookbook provides a resource for managing DNS on Windows hosts.
...
 
+- Windows Server 2008R2
 - Windows Server 2012R2
 - Windows Server 2016
+- Windows Server 2019
 
 ### Chef
 
-- Chef 13
+- Chef 14
```
Comment (tas50) [inline README.md]: Are you actually using something that requires this?

**Comment by Stromweld** [comment]
@tas50 not sure why the integration tests are failing. They pass on my local machine. I was having issues with timing for the second check. The sleep command helped fix it locally.

File: README.md [hunk 3]
```diff
@@ -69,6 +71,34 @@ windows_dns_zone 'chef.local' do
...
+- `search_domains` : Array, Domain suffix to append to short hostnames when searching DNS, this property is global to all interfaces on a server
+
+#### Examples
+
+```ruby
+# Set DNS servers IP and Search suffix for Ethernet 2
+
+windows_dns_client 'default_interface' do
+  interface_alias 'Ethernet 2'
+  search_domains ['chef.local']
```
Comment (xorima) [inline README.md]: tiny thing but shouldn't this be 
```suggestion
  search_domains %w(chef.local)
```

File: kitchen.yml [hunk 4]
```diff
@@ -3,10 +3,12 @@ driver:
...
-    memory: 4096
-    differencing_disk: true
-  network:
-    - ["public_network", {bridge: "Default Switch"}]
+    memory: 2048
+    clipboard: bidirectional
+    draganddrop: bidirectional
+#    differencing_disk: true
+#  network:
+#    - ["public_network", {bridge: "Default Switch"}]
```
Comment (xorima) [inline kitchen.yml]: this was me for hyper-v, could you just remove this for now please? :)

Comment (Stromweld) [reply kitchen.yml]: will do

**Comment by Stromweld** [comment]
These tests are boggling me. Is appveyor possibly overriding my DNS settings changes?

**Comment by Stromweld** [comment]
@Xorima I know you were busy around the time I created this, wondering if you have a chance to look at the test now. They pass on my local machine but don't seem to pass in testing here.

**Pull request closed without merging**